examples/src/bin/listbox_model.rs

```diff
@@ -19,6 +19,96 @@ use std::env::args;
 
 use row_data::RowData;
 
+mod model {
+    use super::*;
+    use gio::subclass::ObjectSubclass;
+    use row_data::RowData;
+    mod imp {
+        use super::*;
+        use gio::subclass::ListModelImpl;
+        use glib::subclass;
+        use glib::subclass::prelude::*;
+        use std::cell::RefCell;
+        #[derive(Debug)]
+        pub struct Model(pub RefCell<Vec<RowData>>);
+
+        // Basic declaration of our type for the GObject type system
+        impl ObjectSubclass for Model {
+            const NAME: &'static str = "Model";
+            type Type = super::Model;
+            type ParentType = glib::Object;
+            type Instance = subclass::simple::InstanceStruct<Self>;
+            type Class = subclass::simple::ClassStruct<Self>;
+
+            glib_object_subclass!();
+
+            // Called right after class_init and allows a GObject to specify
+            // which interfaces it implement, in this case gio::ListModel
```

Comment:
Actually it's called even before `class_init`. On the first call to `ObjectSubclass::get_type()` :)